packages/gatsby-plugin-image/src/components/compat.browser.tsx

```diff
@@ -49,7 +49,7 @@ export function _createCompatLayer(
     Tag,
     ...props
   }) {
-    let rewiredProps: Partial<GatsbyImageProps> = {
+    const rewiredProps: Partial<GatsbyImageProps> = {
       alt: ``,
       as: Tag,
```

Comment:
I would add `image` as an `undefined` property. This is good for perf and it's a good place for describing the property.